x-pack/plugins/security/server/config.ts

```diff
@@ -46,15 +46,13 @@ export const ConfigSchema = schema.object(
     }),
     authc: schema.object({
       providers: schema.arrayOf(schema.string(), { defaultValue: ['basic'], minSize: 1 }),
-      oidc: providerOptionsSchema('oidc', schema.maybe(schema.object({ realm: schema.string() }))),
+      oidc: providerOptionsSchema('oidc', schema.object({ realm: schema.string() })),
       saml: providerOptionsSchema(
         'saml',
-        schema.maybe(
-          schema.object({
-            realm: schema.maybe(schema.string()),
-            maxRedirectURLSize: schema.byteSize({ defaultValue: '2kb' }),
-          })
-        )
+        schema.object({
+          realm: schema.maybe(schema.string()),
+          maxRedirectURLSize: schema.byteSize({ defaultValue: '2kb' }),
+        })
```

Comment:
@elastic/kibana-security FYI for some reason this was different between master and 7.x. I updated this 7.x backport to match master. The last change in master was [here](https://github.com/elastic/kibana/pull/44513/files#diff-909ddd252d4c9db50ce426a70f29168cR43) however the backport still had the `schema.maybe`. Hopefully that was intentional.